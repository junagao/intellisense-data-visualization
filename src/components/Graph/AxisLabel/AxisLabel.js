import React from 'react';
import PropTypes from 'prop-types';

import AxisLabelText from './AxisLabel.styles';

const AxisLabel = ({ x, y, dy, transform, label }) => (
  <AxisLabelText x={x} y={y} dy={dy} transform={transform}>
    {label}
  </AxisLabelText>
);

AxisLabel.propTypes = {
  dy: PropTypes.string,
  transform: PropTypes.string,
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  label: PropTypes.string.isRequired,
};

AxisLabel.defaultProps = {
  dy: null,
  transform: null,
};

export default AxisLabel;
