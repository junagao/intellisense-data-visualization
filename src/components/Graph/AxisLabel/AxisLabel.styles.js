import styled from 'styled-components';

const AxisLabelText = styled.text`
  font-size: 0.85rem;
  opacity: 0.75;
`;

export default AxisLabelText;
