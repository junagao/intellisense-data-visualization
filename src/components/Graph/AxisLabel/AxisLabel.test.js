import React from 'react';
import { shallow } from 'enzyme';

import AxisLabel from './AxisLabel';

describe('AxisLabel', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <AxisLabel
        x={100}
        y={299}
        dy="1rem"
        transform="rotate(-90)"
        label="Value"
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one AxisLabelText element with the correct `x, y, dy, transform` props and show the label', () => {
    expect(wrapper.find('AxisLabelText').length).toEqual(1);
    expect(wrapper.find('AxisLabelText').prop('x')).toEqual(100);
    expect(wrapper.find('AxisLabelText').prop('y')).toEqual(299);
    expect(wrapper.find('AxisLabelText').prop('dy')).toEqual('1rem');
    expect(wrapper.find('AxisLabelText').prop('transform')).toEqual(
      'rotate(-90)',
    );
    expect(wrapper.find('AxisLabelText').at(0).text()).toEqual('Value');
  });

  it('should render one AxisLabelText element with `x, y` props and `dy, transform` default props if they are null', () => {
    wrapper.setProps({ transform: null });
    wrapper.setProps({ dy: null });
    expect(wrapper.find('AxisLabelText').length).toEqual(1);
    expect(wrapper.find('AxisLabelText').prop('x')).toEqual(100);
    expect(wrapper.find('AxisLabelText').prop('y')).toEqual(299);
    expect(wrapper.find('AxisLabelText').prop('dy')).toBeNull();
    expect(wrapper.find('AxisLabelText').prop('transform')).toBeNull();
    expect(wrapper.find('AxisLabelText').at(0).text()).toEqual('Value');
  });
});
