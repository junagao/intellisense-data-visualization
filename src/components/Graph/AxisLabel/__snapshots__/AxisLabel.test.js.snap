// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AxisLabel should render correctly given the required props 1`] = `
<AxisLabelText
  dy="1rem"
  transform="rotate(-90)"
  x={100}
  y={299}
>
  Value
</AxisLabelText>
`;
